### PR TITLE
LGA-644 - Multiple research preferences options

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

- Creates a new ContactResearchMethod model
- Adds ManyToMany field to the PersonalDetails model to create a relationship to the ContactResearchMethod model
- Creates a migration that creates default contact research methods
- Creates a migration to migrate data from previous contact_for_research_via field to the new field.

## Any other changes that would benefit highlighting?
Call centre operators would like to select more than one contact method when contacting the user for research.

https://github.com/ministryofjustice/cla_backend/pull/577 must be merged before this PR
## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/legalaid/migrations/0021_auto_20190515_1042.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                ),
+                ("reference", uuidfield.fields.UUIDField(unique=True, max_length=32, editable=False, blank=True)),
+            ],
+            options={"abstract": False},
+            bases=(core.cloning.CloneModelMixin, models.Model),
+        ),
+        migrations.AddField(
+            model_name="personaldetails",
+            name="contact_for_research_methods",
+            field=models.ManyToManyField(to="legalaid.ContactResearchMethod", null=True),
```
Comment (sldblog) [inline cla_backend/apps/legalaid/migrations/0021_auto_20190515_1042.py]: _curious_
Is this going to be an array field?

Comment (said-moj) [reply cla_backend/apps/legalaid/migrations/0021_auto_20190515_1042.py]: @sldblog Yep it's moving from a single value for contact preference to multiple values

Comment (sldblog) [reply cla_backend/apps/legalaid/migrations/0021_auto_20190515_1042.py]: Sorry, what I meant is that I'm curious how this column type looks in PostgreSQL.

Comment (sldblog) [reply cla_backend/apps/legalaid/migrations/0021_auto_20190515_1042.py]: _not a blocker!_

Comment (exonian) [reply cla_backend/apps/legalaid/migrations/0021_auto_20190515_1042.py]: m2m through table

File: cla_backend/apps/cla_provider/serializers.py [hunk 2]
```diff
@@ -283,6 +284,10 @@ class Meta(AdaptationDetailsSerializerBase.Meta):
         )
 
 
+class ContactResearchMethodSerializer(ContactResearchMethodSerializerBase):
```
Comment (exonian) [inline cla_backend/apps/cla_provider/serializers.py]: What is the purpose of not just using the `Base` class? (and then renaming it to remove the `Base` from the name if it's always used directly and never subclassed)

Comment (said-moj) [reply cla_backend/apps/cla_provider/serializers.py]: Good point. I think I was just copying the pattern for having a separate serialiser for the providers but the fields are the same as the call_centre fields so don't need this empty class. Will remove and update.

**Pull request merged**